source/advanced-topics.adoc

```diff
@@ -408,6 +408,145 @@ an `(opts)` field. The `opts` (the string between parentheses) is passed
 exactly as is to getopt(3) for argc/argv processing at the beginning of
 a macro invocation.
 
+[[setup]]
+=== %setup
+
+With a bunch of source code tarballs used to build the package, the powerful
+``%setup`` macro and all its advanced features might come handy. But lets begin
+with its standard behavior. You can study it from the ``rpmbuild`` output. At
+the beginning of each phase, it outputs ``Executing(%something)``, e.g.:
+
+[source,bash]
+----
+Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.DhddsG
+----
+
+followed by output, similar to the output of shell when ``set -x`` is enabled
+because it is actually shell output with ``set -x`` enabled. If that is not
+enough, it is possible to peek at the content of ``/var/tmp/rpm-tmp.DhddsG``.
+But since ``rpmbuild`` deletes such temporary files after successful build, you
+have to run it with ``--debug`` option. You can see there the setup of
+environment variables followed by something like:
+
+[source,bash]
+----
+cd '/builddir/build/BUILD'
+rm -rf 'cello-1.0'
+/usr/bin/gzip -dc '/builddir/build/SOURCES/cello-1.0.tar.gz' | /usr/bin/tar -xof -
+STATUS=$?
+if [ $STATUS -ne 0 ]; then
+  exit $STATUS
+fi
+cd 'cello-1.0'
+/usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+----
+
+As you can see, the ``%setup`` ensured we are working in the right directory,
+removed residues of previous builds, unpacked the source tarball and set up some
+default privileges. If we want to adjust the behavior of the ``%setup`` macro,
+we can achieve the by various options.
+
+[[setup-q]]
+==== ``%setup -q``
+
+Option ``-q`` limits verbosity of ``%setup`` macro, which mainly means that only
+``tar -xof`` is executed instead of ``tar -xvvof``. This option is listed here
+as first because it has to be used first.
+
+[[setup-n]]
+==== ``%setup -n``
+
+Sometimes, the folder from expanded tarball has a different name then the
+expected ``%{name}-%{version}``. This could confuse the ``%setup`` macro and
```

Comment:
It appears that you are missing a comma before the coordinating conjunction and in a compound sentence. Consider adding a comma.
macro,